### ISSUE TITLE
SelectPicker synthetic event reuse warning on mouse over

### ISSUE THREAD

**Issue by julianrojas87** [rsuite/rsuite#2542]
### What version of rsuite are you using?

5.15.0

### What version of React are you using?

16.14.0

### What browser are you using?

Chrome

### Describe the Bug

Hovering over SelectPickers starting v5.15.0 produces a cascade of synthetic event reuse warnings:
```
Warning: This synthetic event is reused for performance reasons. 
If you're seeing this, you're accessing the property `pageX` on
a released/nullified synthetic event. This is set to null. 
If you must keep the original synthetic event around, use event.persist(). 
See https://fb.me/react-event-pooling for more information.
    in OverlayTrigger (created by Context.Consumer)
    in PickerToggleTrigger (created by SelectPicker)
    in SelectPicker (created by MainLayout)
...
```
One solution is to locally persist the event properties of [this function](https://github.com/rsuite/rsuite/blob/746c2b2cd12a3fcdc46ea6709a3dd25f492b3285/src/Overlay/OverlayTrigger.tsx#L346) as follows:
```ts
const handledMoveOverlay = useCallback(
   (event: React.MouseEvent<Element, MouseEvent>) => {
      const { pageX, pageY, clientX, clientY } = event;
      setCursorPosition(() => ({
         top: pageY,
         left: pageX,
         clientTop: clientX,
         clientLeft: clientY
      }));
   },
   [open]
);
```

### Expected Behavior

No warnings.

### To Reproduce

Can be reproduced on this codesandbox: https://codesandbox.io/s/boring-morning-36gw80?file=/src/index.js

**Comment by himanreddy** [comment]
Facing the same issue after updating to 5.15.1 from 5.4.0
The issue persists even in 5.14.0

**Comment by andrew-y0u** [comment]
The same issue v5.16.1

**Comment by simonguo** [comment]
Fixed in rsuite@5.16.3

**Comment by SevenOutman** [comment]
I'm closing this issue as it seems to have been resolved. Feel free to re-open if it's actually not.
